pkg/vault/test/foundry/BatchRouterNestedPools.t.sol

```diff
@@ -0,0 +1,351 @@
+
+        childPoolA = _createPool([address(usdc), address(weth)].toMemoryArray(), "childPoolA");
+        childPoolB = _createPool([address(wsteth), address(dai)].toMemoryArray(), "childPoolB");
+        parentPool = _createPool(
+            [address(childPoolA), address(childPoolB), address(dai)].toMemoryArray(),
+            "parentPool"
+        );
+
+        vm.startPrank(lp);
+        uint256 childPoolABptOut = _initPool(childPoolA, [poolInitAmount, poolInitAmount].toMemoryArray(), 0);
+        uint256 childPoolBBptOut = _initPool(childPoolB, [poolInitAmount, poolInitAmount].toMemoryArray(), 0);
+
+        uint256[] memory tokenIndexes = getSortedIndexes([childPoolA, childPoolB, address(dai)].toMemoryArray());
+        uint256 poolAIdx = tokenIndexes[0];
+        uint256 poolBIdx = tokenIndexes[1];
+        uint256 daiIdx = tokenIndexes[2];
+
+        uint256[] memory amountsInParentPool = new uint256[](3);
+        amountsInParentPool[daiIdx] = poolInitAmount;
+        amountsInParentPool[poolAIdx] = childPoolABptOut;
+        amountsInParentPool[poolBIdx] = childPoolBBptOut;
+        vm.stopPrank();
+
+        approveForPool(IERC20(childPoolA));
+        approveForPool(IERC20(childPoolB));
+        approveForPool(IERC20(parentPool));
+
+        vm.startPrank(lp);
+        _initPool(parentPool, amountsInParentPool, 0);
+        vm.stopPrank();
+    }
+
+    function testRemoveLiquidityNestedPool__Fuzz(uint256 proportionToRemove) public {
+        // Remove between 0.0001% and 50% of each pool liquidity.
+        proportionToRemove = bound(proportionToRemove, 1e12, 50e16);
+
+        uint256 totalPoolBPTs = BalancerPoolToken(parentPool).totalSupply();
+        // Since LP is the owner of all BPT supply, and part of the BPTs were burned in the initialization step, using
+        // totalSupply is more accurate to remove exactly the proportion that we intend from each pool.
+        uint256 exactBptIn = totalPoolBPTs.mulDown(proportionToRemove);
+
+        NestedPoolTestLocals memory vars = _createNestedPoolTestLocals();
+
+        // During pool initialization, MIN_BPT amount of BPT is burned to address(0), so that the pool cannot be
+        // completely drained. We need to discount this amount of tokens from the total liquidity that we can extract
+        // from the child pools.
+        uint256 deadTokens = (MIN_BPT / 2).mulDown(proportionToRemove);
+
+        uint256[] memory expectedAmountsOut = new uint256[](4);
+        // DAI exists in childPoolB and parentPool, so we expect 2x more DAI than the other tokens.
+        // Since pools are in their initial state, we can use poolInitAmount as the balance of each token in the pool.
+        // Also, we only need to account deadTokens once, since we calculate the bpts in for the parent pool using
+        // totalSupply (so the burned MIN_BPT amount does not affect the bpt in calculation and the amounts out are
+        // perfectly proportional to the parent pool balance)
```

Comment:
```suggestion
        // Also, we only need to account for deadTokens once, since we calculate the BPT in for the parent pool using
        // totalSupply (so the burned MIN_BPT amount does not affect the BPT in circulation, and the amounts out are
        // perfectly proportional to the parent pool balance).
```
Is this what you mean?